Initial Integration

Captainyogurt100 opened the pull request:
This pull request is for the initial integration of GUI, Socket, and Game logic code on master branch. 
- Refactored Start, Connection, and Lobby screens
- Integrated the Game Logic 

Able to Launch GUI accept user input and display user name on the lobby screen.

Commit:
Added ViewController.java

Removed ServerMain.java. No longer needed since Server/Client are a part of single application

Commit:
Deleted Tester class

Class no longer needed

Commit:
Added logic for  player character selection to socket 

When a player selects a character in the lobby screen, a message containing the player id and character selection is sent through the controller to game logic to which it is routed to the server to be processed and sent to the other players as a player update

Commit:
Added character field to Player class

Added character field to player class. Added new constructor with player name and id

Commit:
Added variable use int field

Added variable use int field to BaseMessage class. This field is intended to be a multipurpose int designed to allow different types of information to be store in a message based on the Action enum. Added PLAYER_SELECTION to Action enum

Commit:
Made PlayerMain the application start

PlayerMain contains the main application method

Commit:
Removed main method

Removed main method. added method to return the instance of UserInterface.

Commit:
Refactored PlayerChoiceSpot

Capitalized the class name

Commit:
Added socket/server initiation and lobby creation to GameLogic

Added function to create sockets. Added function to create lobby screen to game logic. Added function to send and receive player character selections

Commit:
Added function to get player object based on id

In PlayerManager, created function to retrieve a player from storage based on player id

Commit:
Fixed spacing issues

Commit:
Slight refactoring

Added comments, reordered function calls to increase readability, updated scene size. Updated ConnectionScreen to accept the UserInterface instance and boolean value as constructor arguments. removed String argument from createScene function for ConnectionScreen

Commit:
Refactored LobbyScreen Class

Used Stack pane to put Player name inside Rectangle Placeholder.  Added Player ArrayList to store all players information. Updated addPlayer function to accept player name, id, color, and if player is root player. Added functions to transform JavaFX.Color to for serializing messages. Added function to add player to Player ArrayList. Using a Player ArrayList makes accessing data to be displayed on the Lobby screen easier. Fixes issue where if other players are connected and a new player is added, all player character selections are reset

Commit:
Refactored connection screen

Added comments, removed dead code. Added error checking for User inputs. Updated Connection screen constructor. Updated connection screen window size. reordered function calls to increase readability.

Commit:
Updated imports

ryanlpeterson commented on src/gui/playerChoiceSpot.java:
```diff
@@ -3,11 +3,11 @@
 
-public class playerChoiceSpot extends Rectangle {
-	
+public class PlayerChoiceSpot extends Rectangle 
```
The filename also needs changed to PlayerChoiceSpot.

ryanlpeterson commented on src/gui/LobbyScreen.java:
```diff
@@ -1,34 +1,78 @@
 package gui;
 
 import java.util.ArrayList;
-import java.util.Arrays;
```
Why was this import removed? Looks like it's still needed for line #42 in new file (within LobbyScreen(int assignedPlayerIndex, String[] allPlayerNames) constructor.

ryanlpeterson commented on src/socket/ClientMessageReceiver.java:
```diff
@@ -126,10 +130,32 @@ public static void sendMessage(Object msg)
+    		processStatusMessage(msg);
+    	}
+
+      GameProcessor.processMessage(msg);
```
Variable msg is not within scope here.

ryanlpeterson commented on src/socket/ClientMessageReceiver.java:
```diff
@@ -126,10 +130,32 @@ public static void sendMessage(Object msg)
      * TODO: Game logic function should be called here. Function that receives Message and processes it.
      * @param msg
```
Javadoc should be adjusted to reflect parameter name change.

ryanlpeterson commented on src/controller/ViewController.java:
```diff
@@ -0,0 +1,72 @@
+import logic.GameProcessor;
+import gui.UserInterface;
+
+public class ViewController 
```
Should we consider ditching this ViewController class? The idea would be for it to abstract/encapsulate the communication between the user's actions and the GameProcessor, but it seems that in most cases the ViewController won't really be encapsulating anything. The current flow would be: UI event handler -> ViewController -> GameProcessor, but the calls to GameProcessor seem like they will be like 1 line in most cases. Should we consider adjusting to: UI event handler -> GameProcessor, and consider our event handlers the "controller" in our MVC pattern?

ryanlpeterson reviewed (changes requested):
I put some comments related to errors I saw when testing. Once these are resolved, I'll approve. The comment I made about the ViewController class could be discussed later if we'd rather merge sooner, but I think it should be discussed soon.

Commit:
Removed obsolete cops

Removed extra GameProcessor and ViewController class. Updated function calls. refactored remaining GameProcessor class

Commit:
updated javadoc
